### ISSUE TITLE
Enable Ruff annotations checks

### ISSUE THREAD

**Issue by patrick91** [strawberry-graphql/strawberry#2477]
In #2476 we enabled a bunch of additional checks on Ruff, but I haven't enabled some that are probably more important for us, namely the ones for annotations. We'd like our library to be fully annotated 😊

We can enable all these checks one at the time:

- [ ] ANN001
- [ ] ANN002
- [x] ANN003
- [ ] ~ANN102~
- [x] ANN201
- [x] ANN202
- [x] ANN204
- [x] ANN205
- [x] ANN206
- [ ] ANN401

We can skip these checks on tests, it's nice to have tests type too, but it might take a lot of time 😊

Feel free to ask me any question for this, some of the annotations might be easy to add, some might not 😊

**Comment by patrick91** [comment]
Annotations are disabled here: https://github.com/strawberry-graphql/strawberry/blob/main/pyproject.toml#L178-L187

To work on any of them, remove the line of the annotation and then run `ruff .` and see what you get 😊 Then you can work from there to add annotations :)

**Comment by MadhuMPandurangi** [comment]
Hi @patrick91, I'm interested in contributing to this issue. Can you please assign this to me
Thank You.

**Comment by patrick91** [comment]
> Hi @patrick91, I'm interested in contributing to this issue. Can you please assign this to me
> 
> Thank You. 

hi @MadhuMPandurangi, I think it's fine to start a PR directly ☺️ 

Please do only one check at the time, enabling all of them at once will require a lot of work ☺️

**Comment by MadhuMPandurangi** [comment]
Thanks for the reply. If this requires a lot of work then can you suggest any other issue which I can work for starters.

**Comment by alexauritt** [comment]
OK, diving into ANN001. @patrick91 no expectations that you'll want to check a WIP but if you want to take a peak and verify I'm on the right path, might save us some time...

https://github.com/strawberry-graphql/strawberry/commit/c81d989a1d9f9a0fba5f93cc9df79263bf39e0cd

**Comment by patrick91** [comment]
@alexauritt yes, that looks good! Feel free to open a draft PR too if you want more feedback on specific lines 😊

**Comment by patrick91** [comment]
@alexauritt it is also fine to have # noqa: ANN001 where it is quite complex to add annotations btw 😊 we can always do this in multiple PRs :)

**Comment by JadHADDAD92** [comment]
I wanna work on ANN201, maybe it's better to make it with multiple PRs as there are 1387 errors, what do you think @patrick91

**Comment by patrick91** [comment]
@JadHADDAD92 yes, good idea!

**Comment by alexauritt** [comment]
tackled ANN001 and ANN002

**Comment by Ambro17** [comment]
Should i choose any annotation and create a new pr?

**Comment by brunodantas** [comment]
May want to remove [ANN102](https://docs.astral.sh/ruff/rules/missing-type-cls/) from the list, since it is being deprecated.
